lib/internal/cluster/shared_handle.js

```diff
@@ -1,46 +1,49 @@
 'use strict';
+const { Map } = primordials;
 const assert = require('internal/assert');
 const dgram = require('internal/dgram');
 const net = require('net');
 
-module.exports = SharedHandle;
-
-function SharedHandle(key, address, port, addressType, fd, flags) {
-  this.key = key;
-  this.workers = [];
-  this.handle = null;
-  this.errno = 0;
-
-  let rval;
-  if (addressType === 'udp4' || addressType === 'udp6')
-    rval = dgram._createSocketHandle(address, port, addressType, fd, flags);
-  else
-    rval = net._createServerHandle(address, port, addressType, fd, flags);
-
-  if (typeof rval === 'number')
-    this.errno = rval;
-  else
-    this.handle = rval;
+class SharedHandle {
+  constructor(key, address, { port, addressType, fd, flags }) {
+    this.key = key;
+    this.workers = new Map();
+    this.handle = null;
+    this.errno = 0;
+    const rval = this.determineRval(addressType,
```

Comment:
This method is only used here, I don't think it's worth moving to a separate function. I think keeping the original conditionals here is best.

Reply:
The previous implement was using `let` and I think if we can restructure the logic to `const` that solidifies the meaning of that variable. Let's say if one goes from top to bottom reading this file, then reaching this `rvalue` variable one can infer that it will be a fixed rval determined by the function.

Reply:
I don't see the use of `let` being a problem, especially given the size of the function and that the assignments happen immediately after the declaration.

Reply:
`let` send a wrong signal on the meaning of variable as it can be changed, but `const` signifies that the value will not be changed and thus can be believed to be the same throughout the scope of the class.